compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -5618,7 +5618,7 @@ private boolean populateFields(IntersectionContext intersectionContext, BRecordT
                                                    intersectionFieldType, newTypeSymbol, lhsRecordField.pos, SOURCE);
             }
 
-            newTypeFields.put(key, new BField(name, null, recordFieldSymbol));
+            newTypeFields.put(key, new BField(name, recordFieldSymbol.pos, recordFieldSymbol));
```

Comment:
It is correct to set the position here. My concern is why we keep the position as a field of `BField`. Normally, we access the position from the symbol. Here as well, when we need to retrieve the position, we need to obtain it from the symbol.

Reply:
When we create the record field symbol of the resultant record symbol of the intersection, we use the position of this `BField`. Alternatively, we can use `lhsRecordField.symbol.pos` as well here.

https://github.com/ballerina-platform/ballerina-lang/blob/9e6acd237f01eb4575e69f137c4f1206ef6ff8f3/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java#L5619

Reply:
Yes. IMO, That is the best way to access the position. Shall we try whether we can remove `pos` in the `BField` without affecting the code?